test/support/utils_sycl.h

```diff
@@ -157,13 +157,27 @@ struct invoke_on_all_hetero_policies
     }
 };
 
+bool has_aspect(const sycl::device& device, sycl::aspect aspect)
+{
+    try
+    {
+        return device.has(aspect);
+    }
+    catch(sycl::exception&)
+    {
+        // the exception indicates that the aspect has not been implemented yet
+    }
+    // let's avoid using a feature that can not be checked for availability
+    return false;
+}
+
 template <typename T, typename TestName>
 void
 test1buffer()
 {
     const sycl::queue& queue = my_queue; // usm and allocator requires queue
 
-#if _PSTL_SYCL_TEST_USM
+    if(has_aspect(queue.get_device(), sycl::aspect::usm_shared_allocations))
```

Comment:
Is it correct that we continue this function in case when this condition is false ?

Reply:
It is correct, because we tests the functionality with the buffers inside such functions in any case.